pyro/ops/gaussian.py

```diff
@@ -54,7 +110,53 @@ def log_density(self, value):
         result = (-0.5) * torch.matmul(self.precision, value.unsqueeze(-1)).squeeze(-1)
         result = result + self.info_vec
         result = (value * result).sum(-1)
-        return result + self.log_normalizer
+
+        # FIXME is this correct?
+        sqrt_P = self.precision.cholesky(upper=True)
+        sqrt_P_u = self.info_vec.unsqueeze(-1).triangular_solve(sqrt_P).solution.squeeze(-1)
+        u_P_u = sqrt_P_u.pow(2).sum(-1)
+        return result + self.log_normalizer - 0.5 * u_P_u
+
+    def condition(self, value):
+        """
+        Condition this Gaussian on a trailing subset of its state.
+        This should satisfy::
+
+            x.condition(y).dim() == x.dim() - y.size(-1)
+        """
+        assert isinstance(value, torch.Tensor)
+        assert value.size(-1) <= self.info_vec.size(-1)
+
+        # FIXME this is bogus but has the right type:
+        n = self.info_vec.size(-1) - value.size(-1)
+        info_vec = self.info_vec[..., :n]
```

Comment:
I guess for conditioning, `info_vec` should be `self.info_vec[..., :n] - self.precision[..., :n, n:].matmul(value.unsqueeze(-1)).squeeze(-1)`

Reply:
The math would look like
```
-0.5 * (a, b)^T @ precision @ (a, b) + (a, b)^T @ info
= -0.5 * a^T @ P_aa @ a - a^T @ P_ab @ b + a^T @ info_a + ...
= -0.5 * a^T @ P_aa @ a + a^T @ (info_a - P_ab @ b) + ...
```
So I think we'll need to adjust log_normalize according to
```
log_normalize += -0.5 * b^T @ P_bb @ b + b^T @ info_b
```

Reply:
Thanks so much @fehiepsi , this worked perfecty!